packages/typespec-client-generator-core/test/decorators.test.ts

```diff
@@ -4743,4 +4743,256 @@ describe("typespec-client-generator-core: decorators", () => {
+      strictEqual(blobNameParam.optional, false);
+      strictEqual(blobNameParam.kind, "method");
+
+      const blobClient = getBlobClient.response;
+
+      strictEqual(blobClient.kind, "client");
+      strictEqual(blobClient.name, "BlobClient");
+      strictEqual(blobClient.initialization.properties.length, 2);
+
+      ok(blobClient.initialization.properties.find((x) => x.kind === "endpoint"));
+      const blobClientBlobInitializationProp = blobClient.initialization.properties.find(
+        (x) => x.name === "blobName"
+      );
+      ok(blobClientBlobInitializationProp);
+      strictEqual(blobClientBlobInitializationProp.kind, "method");
+      strictEqual(blobClientBlobInitializationProp.onClient, true);
+      strictEqual(blobClient.methods.length, 1);
+
+      const download = blobClient.methods[0];
+      strictEqual(download.name, "download");
+      strictEqual(download.kind, "basic");
+      strictEqual(download.parameters.length, 1);
+      strictEqual(download.parameters[0].name, "blobName");
+      strictEqual(download.parameters[0].onClient, true);
+
+      const downloadOp = download.operation;
+      strictEqual(downloadOp.parameters.length, 1);
+      const blobNameOpParam = downloadOp.parameters[0];
+      strictEqual(blobNameOpParam.name, "blobName");
+      strictEqual(blobNameOpParam.correspondingMethodParams.length, 1);
+      strictEqual(blobNameOpParam.correspondingMethodParams[0], blobClientBlobInitializationProp);
+    });
+    it("some methods don't have client initialization params", async () => {
+      await runner.compileWithCustomization(
+        `
+        @service
+        namespace MyService;
+
+        op download(@path blobName: string, @header header: int32): void;
+        op noClientParams(@query query: int32): void;
+        `,
+        `
+        namespace MyCustomizations;
+
+        model MyClientInitialization {
+          blobName: string;
+        }
+
+        @@clientInitialization(MyService, MyCustomizations.MyClientInitialization);
+        `
+      );
+      const sdkPackage = runner.context.sdkPackage;
+      const client = sdkPackage.clients[0];
+      strictEqual(client.initialization.properties.length, 2);
+      const endpoint = client.initialization.properties.find((x) => x.kind === "endpoint");
+      ok(endpoint);
+      const blobName = client.initialization.properties.find((x) => x.name === "blobName");
+      ok(blobName);
+      strictEqual(blobName.clientDefaultValue, undefined);
+      strictEqual(blobName.onClient, true);
+      strictEqual(blobName.optional, false);
+
+      const methods = client.methods;
+      strictEqual(methods.length, 2);
+      const download = methods[0];
+      strictEqual(download.name, "download");
+      strictEqual(download.kind, "basic");
+      strictEqual(download.parameters.length, 2);
+
+      const blobNameParam = download.parameters.find((x) => x.name === "blobName");
+      ok(blobNameParam);
```

Comment:
parameter `blobName` is bump to client, so parameter `blobName` is not the paramter of the `download` method